### PR TITLE
[RFR] Istex link

### PR DESCRIPTION
**Pull request by ThieryMichel**
capture du 2017-04-07 18 31 35

### CODE REVIEW

File: src/app/js/formats/istex/Component.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+const styles = {
+    text: memoize(status => Object.assign({
+        fontSize: '1.5rem',
+        textDecoration: status === REJECTED ? 'line-through' : 'none',
+    })),
+};
+
+export const IstexView = ({ fieldStatus, data, error, field, resource, p: { t } }) => (
+    <span style={styles.text(fieldStatus)}>
+        <span>ISTEX results for {resource[field.name]}:</span>
```
Comment (djhi) [inline src/app/js/formats/istex/Component.js]: useless line imo but if you do add it then make it translatable

File: src/app/js/formats/istex/Component.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    className: null,
+    fieldStatus: null,
+    shrink: false,
+    data: null,
+    error: null,
+};
+
+export default compose(
+    translate,
+    FetchPaginatedDataHOC(fetchIstexData),
```
Comment (djhi) [inline src/app/js/formats/istex/Component.js]: s/FetchPaginatedDataHOC/fetchPaginatedData/

camelcase for functions and we don't need the suffix

File: src/app/js/formats/istex/IstexItem.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    abstract: PropTypes.string,
+};
+
+IstexItemComponent.defaultProps = {
+    abstract: null,
+};
+
+export default compose(
+    withHandlers({
+        handleOpenUrl: ({ fulltext }) => () => window.open(fulltext),
```
Comment (djhi) [inline src/app/js/formats/istex/IstexItem.js]: `window.open` is not handled the same way in all browsers and should be avoided. Try to use an anchor link (maybe with a target set to blank, however this is a bad practice too)

File: src/app/js/formats/istex/Component.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import IstexItem from './IstexItem';
+
+const styles = {
+    text: memoize(status => Object.assign({
+        fontSize: '1.5rem',
+        textDecoration: status === REJECTED ? 'line-through' : 'none',
+    })),
+};
+
+export const IstexView = ({ fieldStatus, data, error, field, resource, p: { t } }) => (
```
Comment (djhi) [inline src/app/js/formats/istex/Component.js]: we use `p: polyglot` everywhere and we should keep it consistant

**Review by djhi — CHANGES_REQUESTED** [review]

File: src/app/js/formats/istex/Component.js [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { List } from 'material-ui/List';
 import translate from 'redux-polyglot/translate';
 import compose from 'recompose/compose';
 
-import FetchPaginatedDataHOC from '../../lib/FetchPaginatedDataHOC';
+import fetchPaginatedDataForComponent from '../../lib/fetchPaginatedDataForComponent';
```
Comment (djhi) [inline src/app/js/formats/istex/Component.js]: why the suffix ? In React, an HOC is necessarily for a component.

**Review by djhi — CHANGES_REQUESTED** [review]

File: src/app/js/formats/istex/Component.js (hunk 5 above)
Comment (djhi) [reply src/app/js/formats/istex/Component.js]: s/fetchPaginatedDataForComponent/fetchPaginatedData/

Comment (ThieryMichel) [reply src/app/js/formats/istex/Component.js]: Because it is in lib and I want to differentiate it from other function or component

Comment (ThieryMichel) [reply src/app/js/formats/istex/Component.js]: and if I call it fetchPaginatedData it would seem to be a service to fetch paginated data, which it is not.

**Pull request merged**